src/DIRAC/Core/DISET/private/BaseClient.py

```diff
@@ -259,17 +259,19 @@ def __discoverExtraCredentials(self):
       self.__extraCredentials = self.kwargs[self.KW_EXTRA_CREDENTIALS]
 
     # Are we delegating something?
+    delegatedID = self.kwargs.get(self.KW_DELEGATED_ID) or self.__threadConfig.getID()
     delegatedDN = self.kwargs.get(self.KW_DELEGATED_DN) or self.__threadConfig.getDN()
     delegatedGroup = self.kwargs.get(self.KW_DELEGATED_GROUP) or self.__threadConfig.getGroup()
+
+    if delegatedID:
+      self.kwargs[self.KW_DELEGATED_ID] = delegatedID
     if delegatedDN:
       self.kwargs[self.KW_DELEGATED_DN] = delegatedDN
-      if not delegatedGroup:
-        result = Registry.findDefaultGroupForDN(delegatedDN)
-        if not result['OK']:
-          return result
-        delegatedGroup = result['Value']
+    if delegatedGroup:
       self.kwargs[self.KW_DELEGATED_GROUP] = delegatedGroup
-      self.__extraCredentials = (delegatedDN, delegatedGroup)
+
+    if delegatedID or delegatedDN:
+      self.__extraCredentials = (delegatedDN or delegatedID, delegatedGroup)
```

Comment:
The `DN` or `ID` to be submitted depends on the authorization method(certificate or token), so one of them will be `None`.
Therefore it is possible to present both (small changes I think), or not to change anything and to add `ID` instead of `DN` to `delegatedDN`, but in this case there can be confusion proceeding from the name of a variable.

Maybe the correct way instead of `ID` or `DN` to put the DIRAC `username`, because `extraCredentials` are allowed only from a `TrustedHost` host that, IIUC, has access to the register and can get `username@group`. And then we can trust if he is `TrustedHost`.

Reply:
I would not advocate at presenting both DN and ID but if both are there, keep the DN for backward compatibility.

Reply:
backward compatibility is a HIGH priority